File: Leetcode/171-excel-sheet-column-number.py
# Leetcode #171 - https://leetcode.com/problems/excel-sheet-column-number/

class Solution:
    def titleToNumber(self, s: str) -> int:
        res = 0
        
        for i in range(len(s)):
            res += math.floor((ord(s[-(i+1)]) - 65 + 1) * math.pow(26, i))
            
        return res

File: Leetcode/171-excel-sheet-column-number.java
/**
 * 
 * Leetcode #171 - https://leetcode.com/problems/excel-sheet-column-number/
 */

class Solution {
    public int titleToNumber(String s) {
        int res = 0;
        int current;

        for (int i = 0; i < s.length(); i++) {
            current = s.charAt(s.length() - i - 1) - 'A' + 1;
            res += current * Math.pow(26, i);
        }

        return res;
    }
}